### ISSUE TITLE
OIDC | Custom group claim value can be string or array

### ISSUE THREAD

**Issue by mayerraphael** [goharbor/harbor#15416]
**Expected behavior and actual behavior:**
If only one group is available in the specified group claim, this claim is of type StringOrUri. Only if two or more groups are available, the claim value becomes an array. This behaviour is explained for the "aud" claim at [RFC751](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3). 

C#s System.IdentityModel.Tokens.Jwt for example uses this logic by default for all claims. Single value = StringOrUri. Multiple Values = Array. 

IdentityProviders like IdentityServer4 also handle claims like this.

The current behavior seems to always expect an array, even if only one group is provided. If i log in via OIDC and only have one group as a StringOrUri value, the custom group claim is ignored. If two or more groups are specified, and the claim value is of type array, the group mapping works.

**Steps to reproduce the problem:**
Add group claim mapping. Only provide one group in the claim. The claim is issued as value "string" from the IdentityServer. Group permissions do not work. Only if two or more groups are present and the claim is of type Array[StringOrUri], even if the additional groups are of no relevance, access to projects that have group permissions now works.

**Versions:**
- harbor version: [2.3.1]

**Comment by reasonerjt** [comment]
@mayerraphael 
May I know what OIDC provider you are using?

It's been a while but I recall I tested against several OIDC providers and the values of groups are all lists in the payload

**Comment by mayerraphael** [comment]
@reasonerjt 
IdentityServer 4.

I can recommend the JWT Builder by Jamie Kurtz where you can replicate the behavior: http://jwtbuilder.jamiekurtz.com/

Single value = string, multiple values with same key = list.

Single value:
![image](https://user-images.githubusercontent.com/25869389/131963462-dee7cddb-0ac5-487b-9f53-becd3de0c91b.png)

Multiple values:
![image](https://user-images.githubusercontent.com/25869389/131963529-f415d0d5-b140-460d-979c-46b9edf9f5a3.png)

**Comment by reasonerjt** [comment]
Thanks @mayerraphael 

I understand this requirement but it may not be a priority. 
When the support for OIDC group was designed, it was determined that the value of the `group claim` in the token has to be an array of strings, and this worked for the OIDC providers that we tested.
 Is it possible to modify the setting of your OIDC to make it a list of strings?

**Comment by mayerraphael** [comment]
@reasonerjt 
No, it is not. Its default behavior for everything that uses the official Microsoft library "System.IdentityModel.Tokens.Jwt".

**Comment by lfrancke** [comment]
FWIW: We're running into the same issue with JumpCloud and I've seen the same behavior with other OIDC providers before.

**Comment by lfrancke** [comment]
I did reach out to JumpCloud support and received this response

> As the current behavior is “working as designed” based on JumpCloud’s existing logic, changing it to a fixed array format would require a functional enhancement through a feature request.
> Additionally confirming with our engineering team, this is a known behavior.
